src/BuiltInTools/AspNetCoreDeltaApplier/StartupHook.cs

```diff
@@ -0,0 +1,142 @@
+
+    public static void Initialize()
+    {
+        Task.Run(async () =>
+        {
+            try
+            {
+                await ReceiveDeltas();
+            }
+            catch (Exception ex)
+            {
+                Log(ex.Message);
+            }
+        });
+    }
+
+    private static List<Action> GetAssembliesReceivingDeltas()
+    {
+        var receipients = new List<Action>();
+
+        foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+        {
+            var customAttributes = assembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+            var deltaReceiverAttributes = customAttributes.Where(a => a.Key == "ReceiveHotReloadDeltaNotification" && !string.IsNullOrEmpty(a.Value));
+            foreach (var deltaReceiverAttribute in deltaReceiverAttributes)
+            {
+                Log($"Attempting to locate receiver {deltaReceiverAttribute.Value} in assembly {assembly}");
+                var type = assembly.GetType(deltaReceiverAttribute.Value!, throwOnError: false);
+                if (type is null)
+                {
+                    Log($"Could not find delta receiver type {deltaReceiverAttribute.Value} in assembly {assembly}.");
+                    continue;
+                }
+
+                if (type.GetMethod("DeltaApplied", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static) is MethodInfo methodInfo)
+                {
+                    var action = methodInfo.CreateDelegate<Action>();
+                    Action safeAction = () =>
+                    {
+                        try 
+                        { 
+                            action(); 
+                        } 
+                        catch (Exception ex) 
+                        { 
+                            Log(ex.ToString()); 
+                        }
+                    };
+
+                    receipients.Add(safeAction);
+                }
+                else
+                {
+                    Log($"Could not find method 'DeltaApplied' on type {type}.");
+                }
+            }
+        }
+
+        return receipients;
+    }
+
+    public static async Task ReceiveDeltas()
+    {
+        Log("Attempting to receive deltas.");
+
+        using var pipeClient = new NamedPipeClientStream(".", "netcore-hot-reload", PipeDirection.InOut, PipeOptions.CurrentUserOnly | PipeOptions.Asynchronous);
+        try
+        {
+            await pipeClient.ConnectAsync(5000);
+            Log("Connected.");
+        }
+        catch (TimeoutException)
+        {
+            Log("Unable to connect to hot-reload server.");
+        }
+
+        List<Action>? receiveDeltaNotifications = null;
+
+        while (pipeClient.IsConnected)
+        {
+            var bytes = new byte[4096];
+            var numBytes = await pipeClient.ReadAsync(bytes);
+
+            var update = JsonSerializer.Deserialize<UpdatePayload>(bytes.AsSpan(0, numBytes));
```

Comment:
Is it guaranteed that an entire payload fits in 4096 bytes? Will deserialize fail if the entire message isn't there?

Reply:
Oh good call. `pipeClient` is an instance of `Stream` which ideally we could simply pass to the DeserializeAsync, but that call would never finish. I suspect I'm not holding the named pipe right. Let me try this again.

Reply:
Maybe use a System.IO.Pipe to read into and parse payload that way.

Reply:
Did you manage to figure this out?